### PR TITLE
feat(Tag): Add onClick prop

### PR DESCRIPTION
**Pull request by thaytharma**
Adds `onClick`-prop to Tag.

It uses the `Button`-component (unstyled) as base.

### CODE REVIEW

**Comment by langz** [comment]
Seems like the visual tests are failing, do we need to generate new snapshots?
https://github.com/dnbexperience/eufemia/runs/4545510995?check_suite_focus=true

`yarn test:screenshots:update` - https://github.com/dnbexperience/eufemia/blob/main/package.json#L28

File: packages/dnb-eufemia/src/components/tag/style/_tag.scss [hunk 1]
```diff
@@ -16,11 +16,40 @@
...
   }
+
   &__text {
-    color: var(--color-black-80);
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+
+  &--clickable {
```
Comment (langz) [inline packages/dnb-eufemia/src/components/tag/style/_tag.scss:34]: For all of this styling, is there a way to reuse what's been done in other "clickable" elements?
Like reusing mixins like `@include hover()`, `@include focus()`, `@include active()`, etc?

Comment (langz) [inline packages/dnb-eufemia/src/components/tag/style/_tag.scss:34]: Something about this styling that makes it look strange in my version of Safari:
image

**Review by langz — CHANGES_REQUESTED** [review]
Awesome work @thaytharma 🚀

File: packages/dnb-eufemia/src/components/tag/style/_tag.scss (hunk 1 above)
Comment (thaytharma) [reply packages/dnb-eufemia/src/components/tag/style/_tag.scss:34]: Thanks for the heads-up! This has been updated 😊

Comment (langz) [reply packages/dnb-eufemia/src/components/tag/style/_tag.scss:34]: https://eufemia.dnb.no/uilib/helpers/classes#sass-and-mixins

Comment (langz) [reply packages/dnb-eufemia/src/components/tag/style/_tag.scss:34]: Looks great now :)

Comment (langz) [reply packages/dnb-eufemia/src/components/tag/style/_tag.scss:34]: Not sure if we should reuse these mixins or do as we did here, adding our own selectors, like `&:hover`

**Review by langz — APPROVED** [review]

File: packages/dnb-eufemia/src/components/tag/style/_tag.scss (hunk 1 above)
Comment (tujoworker) [reply packages/dnb-eufemia/src/components/tag/style/_tag.scss:34]: If we can wait until starting of jan 22 – I would like to have a peer dev session. Basically two topics:

- Overall accessibility with tags and what semantic to use
- Try "one more time" to use the button component – it gives so much for free (like href support with correct sematnics)

Comment (thaytharma) [reply packages/dnb-eufemia/src/components/tag/style/_tag.scss:34]: Sure! 😃  Let's do that @tujoworker 👍

Comment (thaytharma) [reply packages/dnb-eufemia/src/components/tag/style/_tag.scss:35]: @tujoworker 
This PR be re-done to using `<Button variant="unstyled" ../>`. 
Can you do a re-review when you get a chance? 😊

**Review by dinarosv — APPROVED** [review]
Like this component a lot! 🤩  
I'm thinking we might need more information on when to use and when to not use the clickable tag, so no one puts them where a Button would be more appropriate 💭

**Comment by tujoworker** [comment]
Indeed! Very good work! I will have a last check with Theyan on Friday – just to verify that my commits are ok 👍

**Review by tujoworker — APPROVED** [review]
Indeed! Very good work! I will have a last check with Theyan on Friday – just to verify that my commits are ok 👍

**Pull request merged**